packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -4064,6 +4076,13 @@ export class ITxnSubmit extends Op {
 			}
 		}
 
+		// increase Budget when submit application call transaction
+		for (const tx of this.interpreter.currentInnerTxnGroup) {
```

Comment:
I think you can also use the array filter method here as well. Filter by `tx.type === TransactionTypeEnum.APPLICATION_CALL`

get filtered array length and your `this.interpreter.runtime.ctx.budget = MAX_APP_PROGRAM_COST  * filtered array length;`

The same approach we discussed other places.

Reply:
Updated. I will apply this approach when I see a pattern like above.